Uncaught AssertionError

- PHP Depend version: 3.x
- PHP Version: 8.4
- Installation type: composer
- Operating System / Distribution & Version: (Windows 10 / CentOS 7.6 / ...)

### Current Behavior
```
Fatal error: Uncaught AssertionError: assert(is_callable($process->stdout)) in /var/www/vendor/pdepend/pdepend/src/Engine.php:589
Stack trace:
#0 /var/www/vendor/pdepend/pdepend/src/Engine.php(589): assert(false, 'assert(is_calla...')
#1 /var/www/vendor/pdepend/pdepend/src/Engine.php(530): PDepend\Engine->runMultiProcessParse(Object(ArrayIterator), 528, 8)
#2 /var/www/vendor/pdepend/pdepend/src/Engine.php(299): PDepend\Engine->performParseProcess()
#3 /var/www/vendor/phpmd/phpmd/src/Parser.php(97): PDepend\Engine->analyze()
#4 /var/www/vendor/phpmd/phpmd/src/PHPMD.php(221): PHPMD\Parser->parse(Object(PHPMD\Report))
#5 /var/www/vendor/phpmd/phpmd/src/TextUI/Command.php(158): PHPMD\PHPMD->processFiles('src/', Array, Array, Array, Object(PHPMD\Report))
#6 /var/www/vendor/phpmd/phpmd/src/TextUI/Command.php(227): PHPMD\TextUI\Command->run(Object(PHPMD\TextUI\CommandLineOptions), Object(PHPMD\RuleSetFactory))
#7 /var/www/vendor/phpmd/phpmd/bin/phpmd(49): PHPMD\TextUI\Command::main(Array)
#8 /var/www/vendor/bin/phpmd(119): include('/var/www/vendor...')
#9 {main}
  thrown in /var/www/vendor/pdepend/pdepend/src/Engine.php on line 589
```

Error happened after recent changes

--- a/composer.json
+++ b/composer.json
@@ -12,6 +12,7 @@
     "require": {
         "php": ">=8.1",
         "php-64bit": "*",
+        "ext-sockets": "*",
         "fidry/cpu-core-counter": "^1.1",
         "symfony/config": "^5.4 || ^6.0 || ^7.0 || ^8.0",
         "symfony/dependency-injection": "^5.4 || ^6.0 || ^7.0 || ^8.0",

--- a/src/Engine.php
+++ b/src/Engine.php
@@ -72,6 +72,7 @@ use PDepend\Source\Parser\ParserException;
 use PDepend\Util\Cache\CacheFactory;
 use PDepend\Util\Configuration;
 use React\EventLoop\Factory as LoopFactory;
+use React\Stream\ReadableStreamInterface;
 use React\Stream\WritableStreamInterface;
 use RecursiveDirectoryIterator;
 use RecursiveIteratorIterator;
@@ -527,9 +528,9 @@ class Engine
         if ($fileCount > 1) {
             $coreCount = (new CpuCoreCounter())->getCount();
             if ($coreCount > 1) {
-                $this->runMultiProcessParse($files, $fileCount, $coreCount);
-
-                return;
+                if ($this->runMultiProcessParse($files, $fileCount, $coreCount)) {
+                    return;
+                }
             }
         }
 
@@ -575,7 +576,7 @@ class Engine
     /**
      * @param ArrayIterator<int, string> $files
      */
-    private function runMultiProcessParse(ArrayIterator $files, int $fileCount, int $coreCount): void
+    private function runMultiProcessParse(ArrayIterator $files, int $fileCount, int $coreCount): bool
     {
         $processFactory = new ProcessFactory($this->withoutAnnotations);
         $loop = LoopFactory::create();
@@ -586,7 +587,12 @@ class Engine
             $buffers[$proccessNo] = '';
             $process->start($loop);
 
-            assert(is_callable($process->stdout));
+            if (!$process->stdout instanceof ReadableStreamInterface
+                || !$process->stdin instanceof WritableStreamInterface
+            ) {
+                return false;
+            }
+
             $process->stdout->on('data', function (string $chunk) use (&$buffers, $proccessNo, $files, $process): void {
                 $buffers[$proccessNo] .= $chunk;
 
@@ -606,26 +612,24 @@ class Engine
                         $this->parseExceptions[] = $data;
                     }
 
-                    if ($process->stdin instanceof WritableStreamInterface) {
-                        if ($files->valid()) {
-                            $process->stdin->write($files->current() . "\n");
-                            $files->next();
-                        } else {
-                            $process->stdin->end();
-                        }
+                    if ($files->valid()) {
+                        $process->stdin->write($files->current() . "\n");
+                        $files->next();
+                    } else {
+                        $process->stdin->end();
                     }
                     $this->fireEndFileParsing();
                 }
             });
 
-            if ($process->stdin instanceof WritableStreamInterface) {
-                $process->stdin->write($files->current() . "\n");
-                $files->next();
-            }
+            $process->stdin->write($files->current() . "\n");
+            $files->next();
         }
         $loop->run();
         $cache = $this->cacheFactory->create();
         $this->builder->setCache($cache);
+
+        return true;
     }
 
     private function parseFile(PHPTokenizerInternal $tokenizer, string $file): void

